File: src/SPS.Domain/Entities/Banner.cs
using SPS.Domain.Common;

namespace SPS.Domain.Entities;

public class Banner : BaseEntity<long>
{
    public string? Name { get; set; }
    public string? ContentType { get; set; }
    public string? Uri { get; set; }
    public string? LinkUrl { get; set; }
    public string? LinkTarget { get; set; } = "_self";
    public int ClickCount { get; set; }
    public int ViewCount { get; set; }
    public string? Remark { get; set; }
    public int? PositionId { get; set; }

    // Navigation properties
    public BannerPosition? Position { get; set; }
}

File: docs/old/types/banner.ts
// Banner related types based on backend DTOs

export interface BannerResponse {
  id: number;
  name?: string;
  contentType?: string;
  uri?: string;
  linkUrl?: string;
  linkTarget?: '_blank' | '_self';
  clickCount: number;
  viewCount: number;
  remark?: string;
  positionId?: number;
  positionName?: string;
  createdTime: string;
  updatedTime: string;
}

export interface CreateBannerRequest {
  name: string;
  contentType?: string;
  uri: string;
  linkUrl?: string;
  linkTarget?: '_blank' | '_self';
  remark?: string;
  positionId?: number;
}

export interface UpdateBannerRequest {
  name?: string;
  contentType?: string;
  uri?: string;
  linkUrl?: string;
  linkTarget?: '_blank' | '_self';
  remark?: string;
  positionId?: number;
}
